WIP Hot Ranges

Santamaura opened the pull request:
Add hot ranges feature.

Santamaura committed:
wip setup basic redux flow to old hot ranges rpc

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69306)
<!-- Reviewable:end -->

Santamaura committed:
triaging babel compile failure

Santamaura committed:
complete hot ranges table presentational component

Santamaura committed:
update link paths for hot ranges

Santamaura committed:
move hot range files to match new dir structure

Santamaura committed:
styling for table and datetime info

koorosh committed:
server: hot ranges api

This change implements second version of hot ranges api that's required
for UI to represent enhanced Hot Ranges page.

Before, Hot ranges (under Advanced debugging page) used former version of
HotRanges api that provided information about hot ranges with internal/
sensitive data (see: https://github.com/cockroachdb/cockroach/issues/53212)
that should not be exposed to users.

Now, in addition to existing endpoint, additional one is implemented that is
based on a former version and provides hot ranges information that is only
needed for Hot Ranges page (range id, qps, table, db, and index names for
particular range).

The list of hot ranges and their QPS is provided by `HotRanges` service and
then information like DB, table and index names are retrieved from range's
`StartKey` that might include this info or not (in case if it's meta range,
or range that stores index itself for instance).

`HotRange` and `HotRangeV2` services expect the same request type as an
argument but return different responses.
`HotRangeV2` service returns a flat list of hot ranges instead of grouped
ranges per node/store.

Release note: None

koorosh committed:
server: add leaseholder node id to hot ranges api

Current change extends `statuspb.HotRangesResponse` to
include `LeaseholderNodeID` field to indicate the
node id that contains leaseholder replica for
current hot range.

This change was made in `localHotRanges` function (that is
used by `HotRanges` that in turn used by `HotRangeV2` service)
to reuse existing logic of iteration over the stores and querying
hot ranges. It extends its response by `LeaseholderNodeID` field.
Otherwise, the same logic should be implemented in `HotRangeV2`
service by calling `VisitStores` iterator.

Release note: None

koorosh committed:
server: update hot ranges api/v2 version

Before, `listHotRanges` request handler of `apiV2Server` relied
on `HotRanges` service that is now should be replaced by new
`HotRangesV2` implementation.

Current change reuses HotRangeV2 service in `api/v2/ranges/hot`
api. It allows to share the same logic between REST and gRPC
endpoints and gradually migrate to new version of API.

Release note: None

koorosh committed:
server: nit. use struct to group hot ranges data

Before, different hot ranges info was kept in different maps and
with this change, `hotRangeReportMeta` struct is defined to group
hot range data for every range.

Release note: None

koorosh committed:
server: nit. add details comments for hot range V2 functions and proto messages

Provide more detailed descriptions on implemented functions and
proto messages related to Hot Ranges functionality.

Release note: None
